akka-http-core/src/test/scala/akka/http/javadsl/HttpExtensionApiSpec.scala

```diff
@@ -0,0 +1,447 @@
+      val address = waitFor(binding).localAddress
+      sub.cancel()
+    }
+
+    "properly bind a server 3" in {
+      val (_, host, port) = TestUtils.temporaryServerHostnameAndPort()
+      val probe = TestSubscriber.manualProbe[IncomingConnection]()
+      val binding = http.bind(host, port, connectionContext, serverSettings, materializer)
+        .toMat(Sink.fromSubscriber(probe), Keep.left)
+        .run(materializer)
+      val sub = probe.expectSubscription()
+      val address = waitFor(binding).localAddress
+      sub.cancel()
+    }
+
+    "properly bind a server 4" in {
+      val (_, host, port) = TestUtils.temporaryServerHostnameAndPort()
+      val probe = TestSubscriber.manualProbe[IncomingConnection]()
+      val binding = http.bind(host, port, connectionContext, serverSettings, loggingAdapter, materializer)
+        .toMat(Sink.fromSubscriber(probe), Keep.left)
+        .run(materializer)
+      val sub = probe.expectSubscription()
+      waitFor(binding).unbind()
+      sub.cancel()
+    }
+
+    // this cover both bind and single request
+    "properly bind and handle a server with a flow 1" in {
+      val (_, host, port) = TestUtils.temporaryServerHostnameAndPort()
+      val flow: Flow[HttpRequest, HttpResponse, NotUsed] = akka.stream.scaladsl.Flow[HttpRequest]
+        .map(req ⇒ HttpResponse.create())
+        .asJava
+      val binding = http.bindAndHandle(flow, host, port, materializer)
+
+      val (_, completion) = http.outgoingConnection(ConnectHttp.toHost(host, port))
+        .runWith(Source.single(HttpRequest.create("/abc")), Sink.head(), materializer).toScala
+
+      waitFor(completion)
+      waitFor(binding).unbind()
+    }
+
+    "properly bind and handle a server with a flow 2" in {
+      val (_, host, port) = TestUtils.temporaryServerHostnameAndPort()
+      val flow: Flow[HttpRequest, HttpResponse, NotUsed] = akka.stream.scaladsl.Flow[HttpRequest]
+        .map(req ⇒ HttpResponse.create())
+        .asJava
+      val binding = http.bindAndHandle(flow, host, port, connectionContext, materializer)
+
+      val (_, completion) = http.outgoingConnection(ConnectHttp.toHost(host, port))
+        .runWith(Source.single(HttpRequest.create("/abc")), Sink.head(), materializer).toScala
+
+      waitFor(completion)
+      waitFor(binding).unbind()
+    }
+
+    "properly bind and handle a server with a flow 3" in {
+      val (_, host, port) = TestUtils.temporaryServerHostnameAndPort()
+      val flow: Flow[HttpRequest, HttpResponse, NotUsed] = akka.stream.scaladsl.Flow[HttpRequest]
+        .map(req ⇒ HttpResponse.create())
+        .asJava
+      val binding = http.bindAndHandle(flow, host, port, serverSettings, connectionContext, loggingAdapter, materializer)
+
+      val (_, completion) = http.outgoingConnection(ConnectHttp.toHost(host, port))
+        .runWith(Source.single(HttpRequest.create("/abc")), Sink.head(), materializer).toScala
+
+      waitFor(completion)
+      waitFor(binding).unbind()
+    }
```

Comment:
The only difference is the overload of the method called, not sure how to document that without repeating the entire signature?